### DISCUSSION TITLE
current_user is always `flask_login.mixins.AnonymousUserMixin`

### DISCUSSION THREAD

**Discussion by AppleBoiy** [pallets-eco/flask-security#987] [Q&A]
I create a flask app as backend and react as frontend separately and when request to `/login` everything seems to be fine but when I tried to request to protected page when `@auth_required("token", "session")` and `@cross_origin(supports_credentials=True)` decorator its return

```json
{
  "meta": {
    "code": 401
  },
  "response": {
    "errors": [
      "You are not authenticated. Please supply the correct credentials."
    ]
  }
}
```

this is log from my backend
```log
2024-06-05 11:03:07 response True
2024-06-05 11:03:07 current_user <flask_login.mixins.AnonymousUserMixin object at 0xffffaf6437c0>
```

```__form__


class ExtendedLoginForm(LoginForm):
    def validate(self):
        response = super(ExtendedLoginForm, self).validate()

        print("response", response)
        print("current_user", current_user)

        return response

```

**Comment by felipebridge** [comment]
The `AnonymousUserMixin` print isn't actually the bug — `validate()` runs during the login POST itself, before Flask-Security calls `login_user()`, so of course nobody's logged in yet at that line.

The 401 on the protected route afterward is the real problem, and it's almost certainly the session cookie not making it back on the cross-origin request. Check three things: `fetch(url, {credentials: 'include'})` (or `axios.defaults.withCredentials = true`) on the React side; `CORS(app, supports_credentials=True, origins=['http://localhost:3000'])` — needs an explicit origin, `*` doesn't work with credentials; and if frontend/backend are on different ports, `SESSION_COOKIE_SAMESITE='None'` + `SESSION_COOKIE_SECURE=True` (needs HTTPS even in dev). Default `Lax` gets silently dropped on cross-site XHR, which makes it look like an auth bug when it's really just the cookie never arriving.

If you'd rather skip the cookie/CORS dance for a decoupled SPA, `@auth_required('token')` + the `Authentication-Token` header from the login response sidesteps this whole class of bug.
